src/impls/serialize.rs

```diff
@@ -0,0 +1,77 @@
+use crate::Counter;
+
+use std::fmt;
+use std::marker::PhantomData;
+use std::hash::Hash;
+use num_traits::Zero;
+use serde::{Serialize, Deserialize};
+use serde::ser::{SerializeMap, Serializer};
+use serde::de::{Deserializer, Visitor, MapAccess};
+
+
+impl<T, N> Serialize for Counter<T, N> 
+where
+    T: Serialize + Hash + Eq,
+    N: Serialize,
+{
+    fn serialize<S>(&self, serializer:S) -> Result<S::Ok, S::Error>
+    where S: Serializer {
+        let mut map = serializer.serialize_map(Some(self.len()))?;
+        for (k, v) in self.into_iter() {
+            map.serialize_entry(&k, &v)?;
+        }
+        map.end()
```

Comment:
Let's make it explicit that we just defer directly to the inner map:

```suggestion
        self.map.serialize(serializer)
```